src/goto-symex/symex_goto.cpp

```diff
@@ -345,8 +345,7 @@ void goto_symext::merge_goto(
   statet &state)
 {
   // check atomic section
-  if(state.atomic_section_id!=goto_state.atomic_section_id)
-    throw "atomic sections differ across branches";
+  PRECONDITION(state.atomic_section_id == goto_state.atomic_section_id);
```

Comment:
I really think this PRECONDITION would benefit from a description - the throw version was more immediately readable...